be/src/exec/vectorized/aggregator.cpp

```diff
@@ -723,8 +748,37 @@ void Aggregator::_init_agg_hash_variant(HashVariantType& hash_variant) {
         }
         }
     }
+
+    int real_fixed_size = 0;
+    if (_group_by_expr_ctxs.size() > 1) {
+        size_t max_size = 0;
+        bool has_null = false;
+        if (is_group_columns_fixed_size(_group_by_expr_ctxs, _group_by_types, &max_size, &has_null)) {
+            if (max_size < 8 || (!has_null && max_size == 8)) {
+                type = _aggr_phase == AggrPhase1 ? HashVariantType::Type::phase1_slice_fx8
+                                                 : HashVariantType::Type::phase2_slice_fx8;
+            } else if (max_size < 16 || (!has_null && max_size == 16)) {
+                type = _aggr_phase == AggrPhase1 ? HashVariantType::Type::phase1_slice_fx16
+                                                 : HashVariantType::Type::phase2_slice_fx16;
+            }
+            if (!has_null) {
+                real_fixed_size = max_size;
+            }
+        }
+    }
     VLOG_ROW << "hash type is "
              << static_cast<typename std::underlying_type<typename HashVariantType::Type>::type>(type);
     hash_variant.init(type);
-}
+
+#define SET_FIXED_SLICE_HASH_MAP_FIELD(TYPE)                  \
+    if (type == HashVariantType::Type::TYPE) {                \
+        hash_variant.TYPE->real_fixed_size = real_fixed_size; \
+    }
+    SET_FIXED_SLICE_HASH_MAP_FIELD(phase1_slice_fx8);
+    SET_FIXED_SLICE_HASH_MAP_FIELD(phase1_slice_fx16);
+    SET_FIXED_SLICE_HASH_MAP_FIELD(phase2_slice_fx8);
+    SET_FIXED_SLICE_HASH_MAP_FIELD(phase2_slice_fx16);
+#undef SET_FIXED_SLICE_HASH_MAP_FIELD
+
+} // namespace starrocks
```

Comment:
this `// namespace starrocks` is not needed, the `}` wraps the end of a function.

Reply:
It's formatted by our clang-format.